CardScan/Classes/ScanViewController.swift

```diff
@@ -484,16 +488,45 @@ import Vision
             return nil
         }
         
-        // we're assuming that the preview frame is centered in the view, if it's not then
-        // this calculation doesn't work. You'd need to grab references to the previewViewFrame
-        // and the regionOfInterestFrame to figure it out
-        
         // use the full width
-        let width = Double(image.width)
+        let width = CGFloat(image.width)
         // keep the aspect ratio at 480:302
         let height = width * 302.0 / 480.0
-        let cx = Double(image.width) / 2.0
-        let cy = Double(image.height) / 2.0
+        
+        // get device screen size
+        let screen = UIScreen.main.bounds
+        let screenWidth = screen.size.width
+        let screenHeight = screen.size.height
+        
+        // ROI center in Points
+        let regionOfInterestCenterX = self.regionOfInterestLabelFrame.origin.x + self.regionOfInterestLabelFrame.size.width / 2.0
+        
+        let regionOfInterestCenterY = self.regionOfInterestLabelFrame.origin.y + self.regionOfInterestLabelFrame.size.height / 2.0
+        
+        // calculate center of cropping region in Pixels.
+        var cx, cy: CGFloat
+        
+        // Find out whether left/right or top/bottom of the image was cropped before it was displayed to previewView.
+        // The size of the cropped region is needed to map regionOfInterestCenter to the image center
+        let imageAspectRatio = CGFloat(image.width) / CGFloat(image.height)
```

Comment:
Consider adding an assertion to make sure that the previewView is using the scaling style that fits this calculation.

Reply:
Done.
`assert()` used to avoid crashes in production